src/Orleans/Runtime/CallbackData.cs

```diff
@@ -29,8 +30,8 @@ internal class CallbackData : ITimebound, IDisposable
         private TimeSpan timeout;
         private SafeTimer timer;
         private ITimeInterval timeSinceIssued;
-        private static readonly Logger logger = LogManager.GetLogger("CallbackData");
-
+        private readonly Logger logger;
+        private readonly ILoggerFactory loggerFactory;
```

Comment:
Hmm, do we really want to allocate these 2 fields (1 of which it's not entirely trivial to initialize, and the other used to initialize the SafeTimer also in a non-trivial way) on each CallbackData? this may have a perf impact.
Thoughts @ReubenBond @sergeybykov ?

Reply:
I concur - we don't want to allocate a new logger for each `CallbackData`. That would be bad for performance on the hot path.

Reply:
we can change them back to be static and set those two fields on silo start up through a public methods or some sort.

Reply:
similar to what I did with   UnobservedExceptionsHandlerClass.InitLogger

Reply:
synced with @jdom  we are taking the approach of pre-creating a Ilogger, cache it in the caller's scope, and feed it to CallBackData when constructing it.